tools/update-container-image-versions.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+set -eu -o pipefail
+# Updates the image version in the Docker Compose and Kubernetes deployments.
+#
+# Run this script as soon as go.mod has been updated.
+#
+# The image versions are extracted from `go.mod`, with the date field
+# and a hash truncated to seven characters.
+# The date separates the day and the time segment with a "T"
+# and appends "Z".
+#
+# Example:
+#
+#   'go.mod' and 'docker-compose.yml'
+#   github.com/buildbarn/bb-browser v0.0.0-20230906070406-881fd822f75e
+#   github.com/buildbarn/bb-browser         v0.0.0-20230906 070406 -881fd822f75e
+#     browser.image:  ghcr.io/buildbarn/bb-browser:20230906T070406Z-881fd82
+#             differences: T, Z and truncated hash.        ^      ^        ^^^^^
+
+get_image_version() {
+    repo="$1"
+    grep -E "github\.com/buildbarn/${repo}" go.mod \
+        | sed 's,.* v0.0.0-\([0-9]\{8\}\)\([0-9]\{6\}\)-\([0-9a-f]\{7\}\)[0-9a-f]\+,\1T\2Z-\3,'
+}
+
+update_image_version() {
+    new_version="$1"
+    image_name="$2"
+    sed -i "s,\(image: ghcr\.io/buildbarn/${image_name}:\)\S*,\1${new_version}," \
+        docker-compose/docker-compose.yml kubernetes/*.yaml
+}
+
+bb_browser_version=$(get_image_version bb-browser)
```

Comment:
nomenclature remark: I think "get_" in functions is almost always redundant, it is indeed a verb, and the shortest and most general verb. But a function called "image_version" is just as descriptive and reduces line noise.